storage/src/adb/operation.rs

```diff
@@ -68,89 +61,48 @@ impl<K: Array, V: Array> Operation<K, V> {
         "array size too small for commit op"
     );
 
-    /// Create a new operation of the given type.
-    pub fn new(t: Type<K, V>) -> Self {
-        match t {
-            Type::Deleted(key) => Self::delete(key),
-            Type::Update(key, value) => Self::update(key, value),
-            Type::Commit(loc) => Self::commit(loc),
-        }
-    }
-
-    /// Create a new update operation that makes `key` have value `value`.
-    pub fn update(key: K, value: V) -> Self {
-        let mut data = Vec::with_capacity(Self::SIZE);
-        data.push(Self::UPDATE_CONTEXT);
-        data.extend_from_slice(&key);
-        data.extend_from_slice(&value);
-
-        Self {
-            data,
-            _phantom: std::marker::PhantomData,
-        }
-    }
-
-    /// Create a new delete operation that removes any value assigned to `key`.
-    pub fn delete(key: K) -> Self {
-        let mut data = Vec::with_capacity(Self::SIZE);
-        data.push(Self::DELETE_CONTEXT);
-        data.extend_from_slice(&key);
-        data.resize(Self::SIZE, 0);
-
-        Self {
-            data,
-            _phantom: std::marker::PhantomData,
-        }
-    }
-
-    /// Create a new commit operation that indicates the current floor on inactive operations is
-    /// `loc`.
-    pub fn commit(loc: u64) -> Self {
-        let mut data = Vec::with_capacity(Self::SIZE);
-        data.push(Self::COMMIT_CONTEXT);
-        data.extend_from_slice(&loc.to_be_bytes());
-        data.resize(Self::SIZE, 0);
-
-        Self {
-            data,
-            _phantom: std::marker::PhantomData,
+    /// If this is a [Operation::Update] or [Operation::Deleted] operation, returns the key.
+    /// Otherwise, returns None.
+    pub fn to_key(&self) -> Option<&K> {
```

Comment:
Nice. Not having this return Option once we added the Commit operation was definitely an oversight.